octue/mixins/hashable.py

```diff
@@ -0,0 +1,78 @@
+import collections.abc
+import functools
+from blake3 import blake3
+
+
+_HASH_PREPARATION_FUNCTIONS = {
+    str: lambda attribute: attribute,
+    int: str,
+    float: str,
+    type(None): lambda attribute: "None",
+    dict: lambda attribute: str(sorted(attribute.items())),
+}
+
+
+class Hashable:
+
+    ATTRIBUTES_TO_HASH = None
+
+    @classmethod
+    def hash_non_class_object(cls, object_):
```

Comment:
If you used `object_` as the internal attribute instead of the function argument, the api of the function would be slightly cleaner.